### PR TITLE
Merge SeFlow++ and ZOD extraction from HiMo Project

### PR DESCRIPTION
**Pull request by Kin-Zhang**
This pull request introduces several improvements and updates to the OpenSceneFlow codebase, primarily focusing on new methods, documentation, Docker and environment setup, and training workflows. 

**Documentation and Training Workflow Updates from HiMo project:**

* Major reorganization of the `README.md` to clearly separate supervised and self-supervised training instructions, update dataset support (added ZOD), and streamline pretrained weight download links and training commands. Also, the changelog section was removed for clarity. [[1]](diffhunk://#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R14-R22) [[2]](diffhunk://#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5L60-L69) [[3]](diffhunk://#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5L110-R115) [[4]](diffhunk://#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5L137-R146) [[5]](diffhunk://#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5L183-R158) [[6]](diffhunk://#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R170-R224) [[7]](diffhunk://#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5L230-R242) [[8]](diffhunk://#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5L254-R259)
* Updating 1) a new three-frame backbone; 2) seflowppLoss; 3) seflowpp_auto from HiMo (SeFlow++) part. Major reorganization of feed-forward self-supervised methods. Now we comment the ssl-loss config by default, it needs to add `+' with new parameters. The ssl_label can be changed for easy future exploration, etc.
* Training commands now consistently use the `optimizer.lr` parameter and improved loss function specification for all models. [[1]](diffhunk://#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5L137-R146) [[2]](diffhunk://#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5L183-R158) [[3]](diffhunk://#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R170-R224)

**Environment and Installation:**

* Dockerfile updated to install `pytorch3d/torch-scatter` via conda/pip to support running all methods inside Docker.
* Revert the environment to python=3.8, torch=2.0.0 and cuda=11.7 check more discussion for upgrading-issue in discussion #11 

---

Old Message:

HiMo got accepted into T-RO now! 🔥 I will try my best to update all detailed codes so that everyone can reproduce all the tables/results/figures in the paper.
All Scene Flow-related parts will be merged into OpenSceneFlow as this pull request shows, and then we will update another repo for the HiMo task and downstream task codes.

- [x] update seflow++ scripts.
- [x] update av2 extraction code. lidar_id and lidar_dt etc.
- [x] update seflow++ weights, seflow repo readme also update.
- [x] update opensf demo data.
- [x] update all baselines (NSFP, FastNSF, ICP-Flow), check these on #15 and #14 
- [x] update ZOD data extraction.

The self-supervised learning SSL part has a big refactor. We added auto_label involved in cfg, etc, for afterward potential seflowpp hyper-paramters tuning by users, etc.

### CODE REVIEW

File: train.py [hunk 1]
```diff
@@ -140,7 +141,8 @@ def main(cfg):
 
     # NOTE(Qingwen): search & check: def training_step(self, batch, batch_idx)
     trainer.fit(model, train_dataloaders = train_loader, val_dataloaders = val_loader, ckpt_path = cfg.checkpoint)
-    wandb.finish()
+    if cfg.wandb_mode != "disabled":
+        wandb.finish()
```
[the remark this answers is not in the document]
Comment (Kin-Zhang) [reply train.py:145]: didn't get this point.... check later.

File: dataprocess/extract_zod.py [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+        break
+    
+    # comment out if you want to process all scene.
+    # if nproc <= 1:
+    #     for x in tqdm(args, ncols=120):
+    #         proc(x, ignore_current_process=True)
+    # else:
+    #     with Pool(processes=nproc) as p:
+    #         res = list(tqdm(p.imap_unordered(proc, args), total=len(logs), ncols=120))
+
```
[the remark this answers is not in the document]
Comment (Kin-Zhang) [reply dataprocess/extract_zod.py:122]: it's ok I think, we already said it on README also. It's mainly for people to speed-up process and check purpose.

**Review by iyuner — APPROVED** [review]

**Comment by Kin-Zhang** [comment]
SeFlow++ ckpt for av2 leaderboard (local results after fixed gt #5 ):

image
